zerver/data_import/slack.py

```diff
@@ -898,6 +899,38 @@ def get_messages_iterator(
         yield from sorted(messages_for_one_day, key=get_timestamp_from_message)
 
 
+def get_parent_user_id_from_thread_message(thread_message: ZerverFieldsT, subtype: str) -> str:
+    """
+    This retrieves the user id of the sender of the original thread
+    message.
+    """
+    if subtype == "thread_broadcast":
+        return thread_message["root"]["user"]
+    elif thread_message["thread_ts"] == thread_message["ts"]:
+        # This is the original thread message
+        return thread_message["user"]
+    else:
+        return thread_message["parent_user_id"]
+
+
+def get_zulip_thread_topic_name(
+    message_content: str, thread_ts: datetime, thread_counter: dict[str, int]
+) -> str:
+    """
+    The topic name format is date + message snippet + counter.
+
+    e.g "2024-05-22 Hello this is a long message that will be c… (1)"
+    """
+    thread_date = thread_ts.strftime(r"%Y-%m-%d")
+    thread_topic_snippet_length = 48 + len(thread_date)
```

Comment:
We should compute 48.

Reply:
Or actually we don't need to if we use `truncate_content` properly, see https://github.com/zulip/zulip/pull/35151/files#r2211627179